### PR TITLE
Data upload example

### PR DESCRIPTION
**Pull request by zhenik**
- [x] Data upload to minio
- [x] Automate table creation via presto-cli

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,8 +65,29 @@ proxy-h:
...
-proxy-p:
-	consul connect proxy -service postgres-local -upstream presto:8080 -log-level debug
+proxy-presto:
+	consul connect proxy -service presto-local -upstream presto:8080 -log-level debug
+
+download-presto-cli:
+	curl https://repo1.maven.org/maven2/io/prestosql/presto-cli/${PRESTO_VERSION}/presto-cli-${PRESTO_VERSION}-executable.jar -o ./${presto}
+	chmod +x ./${presto}
+
+presto-cli:
```
Comment (Neha-Sinha2305) [inline Makefile:79]:
```suggestion
presto-cli: proxy-presto
```

File: Makefile [hunk 2]
```diff
@@ -64,8 +65,29 @@ proxy-h:
...
+ifeq (,$(wildcard ./${presto}))
+	$(MAKE) download-presto-cli
+endif
+	${presto} --server localhost:8080 --catalog hive --schema default --user presto
+
+# manual table creation for different file types
+csv:
```
Comment (Neha-Sinha2305) [inline Makefile:86]: Shouldn't *csv*, *json*, *avro* targets be dependent on *proxy-presto* and *presto-cli* targets?
```suggestion
csv: proxy-presto presto-cli
```

File: dev/ansible/05_presto_create_tables.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  until: result_presto.json | length == 3
+  tags: example-upload
+
+- name: Start local proxy on port 8888 to presto service
+  shell: nohup consul connect proxy -service presto-local -upstream presto:8888 -log-level debug </dev/null >/dev/null 2>&1 &
+  register: test_output
+  tags: example-upload
+
+# To quickly verify that proxy available (without any credentials)
+- name: Minio curl example. Verify that proxy to minio available
```
Comment (Neha-Sinha2305) [inline dev/ansible/05_presto_create_tables.yml:20]:
```suggestion
- name: Presto curl example. Verify that proxy to presto available
```

**Review by Neha-Sinha2305 — CHANGES_REQUESTED** [review]

File: dev/ansible/05_presto_create_tables.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  tags: example-upload
+
+- name: Create JSON table
+  shell: docker run --network host -v "/vagrant/example/resources/query/json_create_table.sql:/json_create_table.sql" fredrikhgrelland/presto-cli ./presto --server localhost:8888 --catalog hive --schema default --user presto --file /json_create_table.sql
+  register: docker_output_cmd_json
+  tags: example-upload
+
+- name: Create AVRO table
+  shell: docker run --network host -v "/vagrant/example/resources/query/avro_tweets_create_table.sql:/avro_tweets_create_table.sql" fredrikhgrelland/presto-cli ./presto --server localhost:8888 --catalog hive --schema default --user presto --file /avro_tweets_create_table.sql
+  register: docker_output_cmd_json
```
Comment (Neha-Sinha2305) [inline dev/ansible/05_presto_create_tables.yml:49]:
```suggestion
  register: docker_output_cmd_avro
```

**Review by Neha-Sinha2305 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (zhenik) [reply Makefile:79]: Proxy-presto stdout is attached to the terminal. I would suggest running these two commands `presto-cli` and `proxy-presto` in separate terminals.

File: Makefile (hunk 2 above)
Comment (zhenik) [reply Makefile:86]: Tables creation is automated via ansible. These commands are manual examples. And again, proxy's stdout is attached to the terminal. I would suggest running `presto-cli` in separate terminal.

Comment (Neha-Sinha2305) [reply Makefile:86]: I guess we should move them out from **Makefile** and add it in the documentation -> example commands for manual table creation for different file types

File: dev/ansible/05_presto_create_tables.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    url: http://localhost:8500/v1/health/checks/presto?filter={{ '(Status=="passing")' | urlencode }}
+    method: GET
+    return_content: yes
+    status_code: 200
+    body_format: json
+  register: result_presto
+  retries: 15
+  delay: 15
+  until: result_presto.json | length == 3
+  tags: example-upload
```
Comment (pdmthorsrud) [inline dev/ansible/05_presto_create_tables.yml:12]:
```suggestion
- name: Presto healthchecks pass
  uri:
    url: http://localhost:8500/v1/health/checks/presto
    method: GET
    return_content: yes
    status_code: 200
    body_format: json
  register: result_presto
  retries: 15
  delay: 15
  until: '"critical" not in result_presto | string'
  tags: example-upload
```

Could we do something like this? More dynamic

**Review by Neha-Sinha2305 — APPROVED** [review]

File: example/resources/schema/Makefile [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+encode-json-2-avro:
+	avro-tools fromjson --codec snappy --schema-file ./avro/twitter_scheme.avsc ./avro/tweet1.json > ../data/avro-tweet/tweet1.snappy.avro
+	avro-tools fromjson --codec snappy --schema-file ./avro/twitter_scheme.avsc ./avro/tweet2.json > ../data/avro-tweet/tweet2.snappy.avro
+
+# protofub
+encode-text-2-proto:
+	mkdir cpp | true
+	protoc --cpp_out=cpp ./proto/info.proto | true
+	protoc --encode=Info ./proto/info.proto < ./proto/info1.text  > ../data/proto-message/info1.data
+	protoc --encode=Info ./proto/info.proto < ./proto/info2.text  > ../data/proto-message/info2.data
```
Comment (pdmthorsrud) [inline example/resources/schema/Makefile:11]: Maybe add a "Required software" section? This is not standard on my machine at least

**Review by claesgill — APPROVED** [review]
Was not able to test, but LGTM 🚀

File: dev/ansible/05_presto_create_tables.yml (hunk 5 above)
Comment (zhenik) [reply dev/ansible/05_presto_create_tables.yml:12]: I would like to stay explicit and not dynamic. 3 healthchecks should have status - passing

Comment (pdmthorsrud) [reply dev/ansible/05_presto_create_tables.yml:12]: I don't have too much of a preference, so that's completely fine with me.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,281 @@
...
+ <a href="https://github.com/fredrikhgrelland/vagrant-hashistack/releases">
+   <img alt="Releases" src="https://img.shields.io/badge/dynamic/json?label=with%20vagrant-hashistack&query=%24.current_version.version&url=https%3A%2F%2Fapp.vagrantup.com%2Fapi%2Fv1%2Fbox%2Ffredrikhgrelland%2Fhashistack"/>
+ </a>
+</p>
 
+#
+
+## Contents
+1. [Compatibility](#compatibility)
+2. [Usage](#usage)
```
Comment (Neha-Sinha2305) [inline README.md:13]: Wondering if we should have the prerequisites section in every tearraform-nomad-module readme that should have a link to https://github.com/fredrikhgrelland/vagrant-hashistack-template#install-prerequisites
Mainly with regarsd to running the 'make install' command as the first command, if the user has not run that on their machine before.

Comment (pdmthorsrud) [reply README.md:13]: Hm, that is a good point. I agree

File: example/resources/schema/Makefile (hunk 6 above)
Comment (zhenik) [reply example/resources/schema/Makefile:19]: All the encoded data is already in the directory `data/` and in running example there is no need to do anything.

These Makefile commands are just an example of how to encode data using cli tools. 
I added a description in readme and error with a link where to download cli tools. Installation of these CLI tools is not necessary until you want to experiment with serialization/deserialization (encoding/decoding).

Comment (pdmthorsrud) [reply example/resources/schema/Makefile:19]: Got it. LGTM

**Review by pdmthorsrud — APPROVED** [review]
LGTM

File: README.md (hunk 7 above)
Comment (zhenik) [reply README.md:14]: Added

**Pull request merged**